cmd/api/logging/logging.go

```diff
@@ -0,0 +1,178 @@
+// an error if there isn't available
+func GetKubeArchiveLoggingCredentials() (map[string]string, error) {
+	loggingDir, exists := os.LookupEnv(files.LoggingDirEnvVar)
+	if !exists {
+		return nil, fmt.Errorf("environment variable not set: %s", files.LoggingDirEnvVar)
+	}
+	configFiles, err := files.FilesInDir(loggingDir)
+	if err != nil {
+		return nil, fmt.Errorf("could not read logging credentials: %w", err)
+	}
+	if len(configFiles) == 0 {
+		return nil, fmt.Errorf("logging secret is empty. To configure logging update the kubearchive-logging Secret")
+	}
+
+	loggingConf, err := files.LoggingConfigFromFiles(configFiles)
+	if err != nil {
+		return nil, fmt.Errorf("could not get value for logging credentials: %w", err)
+	}
+	return loggingConf, nil
+}
+
+// SetLoggingCredentials sets the user and password for accessing logging backend in the request context
+func SetLoggingCredentials(loggingCreds map[string]string, loggingCredsErr error) gin.HandlerFunc {
+	user := loggingCreds[userKey]
+	password := loggingCreds[passwordKey]
+	if user == "" || password == "" || loggingCredsErr != nil {
+		loggingCredsErr = fmt.Errorf(
+			"logging secret values unset. To configure logging update the kubearchive-logging Secret")
+		return func(c *gin.Context) {
+			abort.Abort(c, loggingCredsErr, http.StatusBadRequest)
+		}
+	}
+	return func(c *gin.Context) {
+		c.Set(userKey, user)
+		c.Set(passwordKey, password)
+	}
+}
+
+// LogRetrieval retrieves a middleware function that checks the logging config and when set
+// expects to find a log url in the context from where retrieve and parse logs
+// It should be called when user, password and logURL are set in the context.
+func LogRetrieval() gin.HandlerFunc {
+	// FIXME For now the queries to the logging backend server are done insecurely. Needed for the current test env.
+	// #nosec G402
+	client := http.Client{
+		Transport: otelhttp.NewTransport(&http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true}})}
+	return func(c *gin.Context) {
+		user := c.GetString(userKey)
+		password := c.GetString(passwordKey)
+		if user == "" || password == "" {
+			loggingCredsErr := fmt.Errorf(
+				"unexpected error. Logging credentials are unset")
+			abort.Abort(c, loggingCredsErr, http.StatusInternalServerError)
+			return
+		}
+		logUrl := c.GetString("logURL")
+		jsonPath := c.GetString("jsonPath")
+
+		if logUrl == "" {
+			abort.Abort(c, fmt.Errorf("no log URL found"), http.StatusNotFound)
+			return
+		}
+		slog.Info("Retrieving logs", "logURL", logUrl)
+
+		var jsonPathParser jp.Expr
+		var errJsonPath error
+
+		if jsonPath != "" {
+			jsonPathParser, errJsonPath = jp.ParseString(jsonPath)
+			if errJsonPath != nil {
+				abort.Abort(c, fmt.Errorf("invalid jsonPath %s for url %s", jsonPath, logUrl),
+					http.StatusInternalServerError)
+				return
+			}
+		}
+
+		request, errReq := http.NewRequest(http.MethodGet, logUrl, nil)
```

Comment:
Since your `http.Client` is using `otelhttp.Transport` you should use `http.NewRequestWithContext` so that the Otel trace is passed along.

```suggestion
		request, errReq := http.NewRequestWithContext(c.Request.Context(), http.MethodGet, logUrl, nil)
```

Reply:
Done!